doc/barclamp.md

```diff
@@ -168,14 +181,35 @@ function reposync_crowbar()
       echo "in ${BARCLAMP_FORK_PATH}." 1>&2
       return
     fi
-  else
-    commit="${COMMIT}^"
   fi
 
   pushd $BARCLAMP_FORK_PATH > /dev/null
-  cp -a /opt/dell /opt/dell.orig
-  git diff $commit HEAD | patch -N --merge -p1 -d /opt/dell || return
-  rm -rf /opt/dell.orig
+
+  # This backup preserves the original /opt/dell from the RPM
+  if [ ! -d /opt/dell.orig ]; then
+    cp -a /opt/dell /opt/dell.orig
+  fi
+
+  # This backup preserves the /opt/dell from before the current
+  # reposync_crowbar run. It is only preserved if reposync_crowbar
+  # fails.
+  if [ ! -d /opt/dell.bak ]; then
+    cp -a /opt/dell /opt/dell.bak
+  else
+    echo "Found a previous intermediate backup of /opt/dell in /opt/dell.bak" 1>&2
+    echo "This indicates a failed reposync_crowbar run." 1>&2
+    echo "Please restore /opt/dell to a known good state from either of" 1>&2
+    echo "    /opt/dell.bak" 1>&2
+    echo " or /opt/dell.orig" 1>&2
+    echo " and remove /opt/dell/back.bak once you have done so." 1>&2
+    return 1
+  fi
+  if [ -n "$commit" ]; then  # only apply commits added since the last reposync_crowbar run
```

Comment:
Might be worth a comment that this can only get set via `/opt/dell/patchlevel` now.